js/sign-up.js

```diff
@@ -0,0 +1,63 @@
+const container = document.querySelector('.container');
+const form = document.querySelector('#form');
+const formControl = document.querySelectorAll('.form-control');
+const username = document.querySelector('#username');
+const email = document.querySelector('#email');
+const password = document.querySelector('#password');
+const checkPassword = document.querySelector('#checkpassword');
+
+form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    checkInputs();
+})
+
+function checkInputs() {
+    const usernameValue = username.value.trim();
+    const emailValue = email.value.trim();
+    const passwordValue = password.value.trim();
+    const checkPasswordValue = checkPassword.value.trim();
+
+    if(usernameValue === ""){
+        setErrorFor(username, "Please enter a username");
+    } else {
+        setSuccessFor(username)
+    }
+
+    if(emailValue === ""){
+        setErrorFor(email, "Please enter an email");
+    } else if(!isEmail(emailValue)) {
+        setErrorFor(email, "Please enter a valid email")
+    } else {
+        setSuccessFor(email)
+    }
+
+    if(passwordValue === ""){
+        setErrorFor(password, "Please enter a password");
+    } else {
+        setSuccessFor(password)
+    }
+
+    if(checkPasswordValue === "") {
+        setErrorFor(checkPassword, "Please enter a same password");
+    } else if(checkPasswordValue !== passwordValue) {
+        setErrorFor(checkPassword, "Password does not match");
+    } else {
+        setSuccessFor(checkPassword)
+    }
+}
```

Comment:
no need to validate these input fields in js, these can be validated in HTML itself.